game-app/game-core/src/main/java/games/strategy/triplea/ai/pro/ProCombatMoveAi.java

```diff
@@ -885,40 +888,61 @@ private void determineUnitsToAttackWith(
       }
 
       // Check to see if any territories can be bombed
+      final Map<Territory, BombedTerritoryData> bombedTerritoryDataMap = new HashMap<>();
       final Map<Unit, Set<Territory>> bomberMoveMap =
           territoryManager.getAttackOptions().getBomberMoveMap();
       for (final Unit unit : bomberMoveMap.keySet()) {
         if (alreadyAttackedWithUnits.contains(unit)) {
           continue;
         }
+        final Predicate<Unit> bombingTargetMatch =
+            Matches.unitCanProduceUnitsAndCanBeDamaged()
+                .and(Matches.unitIsLegalBombingTargetBy(unit));
         Optional<Territory> maxBombingTerritory = Optional.empty();
         int maxBombingScore = MIN_BOMBING_SCORE;
         for (final Territory t : bomberMoveMap.get(unit)) {
-          final boolean canBeBombedByThisUnit =
-              t.getUnitCollection()
-                  .anyMatch(
-                      Matches.unitCanProduceUnitsAndCanBeDamaged()
-                          .and(Matches.unitIsLegalBombingTargetBy(unit)));
+          final boolean canBeBombedByThisUnit = t.getUnitCollection().anyMatch(bombingTargetMatch);
           final boolean canCreateAirBattle =
               Properties.getRaidsMayBePreceededByAirBattles(data.getProperties())
                   && AirBattle.territoryCouldPossiblyHaveAirBattleDefenders(t, player, data, true);
           if (canBeBombedByThisUnit
               && !canCreateAirBattle
               && canAirSafelyLandAfterAttack(unit, t)) {
-            final int noAaBombingDefense =
-                t.getUnitCollection().anyMatch(Matches.unitIsAaForBombingThisUnitOnly()) ? 0 : 1;
+            // get territory data that is independent of the attack unit
+            final BombedTerritoryData bombedTerritoryData =
+                bombedTerritoryDataMap.computeIfAbsent(
```

Comment:
Continuing my discussion from the issue.

I don't think this is correct., because bombedTerritoryDataMap isn't keyed by bomber type or factory type, which are relevant for the purpose of `t.getUnitCollection().getMatches(bombingTargetMatch);`.

So you're computing a `minDamageNeeded` value for the territory as a whole which is not right when different units can bomb different types of factories.

As I said earlier, I think the issue goes away if you get rid of the map. Just compute your logic directly in the body, instead of using bombedTerritoryDataMap.computeIfAbsent().

Reply:
As mentioned in the forum I have just committed another commit to introduce `minDamageNeeded vs. maxDamage calculated/aggregated per target unit`.
@asvitkine Please have a look and let me know your thoughts.